platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java

```diff
@@ -573,50 +576,80 @@ public void onSaveInstanceState(@NonNull Bundle outState) {
     }
 
     /**
-     * You must call this method from the parent's {@link Activity#onDestroy()} or {@link Fragment#onDestroy()}.
+     * You must call this method from the parent's {@link Activity#onStart()} or {@link Fragment#onStart()}
      */
     @UiThread
-    public void onDestroy() {
-        destroyed = true;
-        nativeMapView.terminateContext();
-        nativeMapView.terminateDisplay();
-        nativeMapView.destroySurface();
-        nativeMapView.destroy();
-        nativeMapView = null;
+    public void onStart() {
+        onStartCalled = true;
+
+        // Register for connectivity changes
+        connectivityReceiver = new ConnectivityReceiver();
+        getContext().registerReceiver(connectivityReceiver, new IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION));
+
+        nativeMapView.update();
+        myLocationView.onStart();
+
+        // In case that no style was set or was loaded through MapboxMapOptions
+        if (!styleWasSet) {
+            setStyleUrl(styleUrl);
+        }
+    }
+
+    /**
+     * You must call this method from the parent's {@link Activity#onResume()} or {@link Fragment#onResume()}.
+     */
+    @UiThread
+    public void onResume() {
+        if (!onStartCalled) {
+            // TODO: 26/10/16, can be removed after 5.0.0 release
+            throw new IllegalStateException("MapView#onStart() was not called. " +
+                    "You must call this method from the parent's {@link Activity#onStart()} or {@link Fragment#onStart()}.");
+        }
     }
 
     /**
      * You must call this method from the parent's {@link Activity#onPause()} or {@link Fragment#onPause()}.
      */
     @UiThread
```

Comment:
I implemented this way to make sure users are correctly migrating from an older version. If a user doesn't implement the new onStart/onStop, they will not be able to receive any location updates. This is something that will go unnoticed during "migration" and will only show when you are trying to a location fix. I have no problem with removing this but I felt it was an improvement over the other possible UX here.